cmd/oras/internal/option/platform.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/registry"
+)
+
+var (
+	errMediatypeUnsupported = errors.New("Unsupported media type")
+	errNoMatchFound         = errors.New("No matched platform found")
+)
+
+// Platform option struct.
+type Platform struct {
+	Platform string
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Platform) ApplyFlags(fs *pflag.FlagSet) {
+	fs.StringVarP(&opts.Platform, "platform", "", "", "fetch the manifest of a specific platform if target is multi-platform capable")
+}
+
+// parse parses the input platform flag to an oci platform type.
+func (opts *Platform) parse() (ocispec.Platform, error) {
+	var p ocispec.Platform
+	parts := strings.SplitN(opts.Platform, ":", 2)
+	if len(parts) == 2 {
+		// OSVersion is splitted by comma
+		p.OSVersion = parts[1]
+	}
+
+	parts = strings.Split(parts[0], "/")
+	if len(parts) < 2 || len(parts) > 3 {
+		return ocispec.Platform{}, fmt.Errorf("failed to parse platform '%s': expected format os/arch[/variant]", opts.Platform)
+	}
+
+	// OS/Arch/[Variant]
+	p.OS = parts[0]
+	if p.OS == "" {
+		return ocispec.Platform{}, fmt.Errorf("invalide platform: OS cannot be empty")
+	}
+	p.Architecture = parts[1]
+	if p.Architecture == "" {
+		return ocispec.Platform{}, fmt.Errorf("invalide platform: Architecture cannot be empty")
+	}
+	if len(parts) > 2 {
```

Comment:
Yes it is. It will be regarded as use want a image with OS as `os` and architecture as `arch`. BTW It's actually in the `TestPlatform_parse` test function.